### DISCUSSION TITLE
[Question][Hw4] Coverage

### DISCUSSION THREAD

**Discussion by john-mai-2605** [prosyslab-classroom/cs348-information-security#567] [Q&A]
Is the coverage in the evaluation the cumulative coverage?
I read in another issue, you answer that we can check from .cov file.
But .cov file (from my understanding) only provides the last run's coverage.
I guess we should use other way (like cardinality) to get number of lines covered?

**Comment by ee12ha0220** [comment]
I wanted to ask the same thing.

**Comment by kimjungwow** [comment]
The professor said **"We will consider the union of all coverage during the fuzzing, not the latest cov."** in https://github.com/prosyslab-classroom/cs448-2021-spring/issues/102#issuecomment-841751566

**Comment by KihongHeo** [comment]
Thanks @kimjungwow.
TAs will do the followings for evaluation:
- TAs will run your fuzzer with the option to produce all passing inputs
- After timeout, TAs rerun the instrumented binary with all generated inputs and collect all *.cov files
- Compute all cumulated coverage

Make sense?

**Comment by ee12ha0220** [comment]
Thank you! But I have another question: when I give seed_2.json to the json_parser, my produced coverage has 6162 lines. It is bigger then 800, but I think I will not get full points for this, and could you explain this situation?? Thank you

**Comment by mrpositron** [comment]
You probably have duplicate lines. Try to filter them out.

**Comment by ee12ha0220** [comment]
Thanks :)

**Comment by john-mai-2605** [comment]
Got it. Thank you.
I ll leave this one open since other might have the same question

**Comment by Standchen** [comment]
@KihongHeo 

> After timeout, TAs rerun the instrumented binary with all generated inputs and collect all *.cov files

Does `all generated inputs` mean by all the passing & **crahsed** inputs? or just passing inputs?

**Comment by KihongHeo** [comment]
All passing & crashing inputs.

**Comment by john-mai-2605** [comment]
Closed
